lib/OpenQA/Schema/ResultSet/JobSettings.pm

```diff
@@ -68,4 +68,14 @@ sub query_for_settings ($self, $args) {
     return $self->search({-and => \@conds});
 }
 
+sub parent_worker_class ($self, $parent_job_id, $worker_class) {
+    my $result = $self->search({parent_job_id => $job_id, key => 'WORKER_CLASS'})->next;
+    my $parent_worker_class = $result ? $result->value : '';
+    die
+      "Worker class $worker_class does not match the worker class of its directly chained parent ($parent_worker_class)"
+      unless $worker_class eq $parent_worker_class;
+    die "Ambiguous job settings for dependent jobs" if $result->next;
```

Comment:
We may have multiple `WORKER_CLASS` settings per job and that's perfectly fine. This code should actually get all of them and then compare whether the full set of worker classes is identical to the full set of worker classes of the parent job (the order doesn't matter).

Note that this means that already the line `$job->settings->find({key => 'WORKER_CLASS'});` is problematic. Also in this case we need to handle that there can be multiple worker classes (and then compare it to the multiple worker classes the call you're trying to fix might return).

Reply:
Right, like I wrote. And in the case of `ResultSet::Jobs::create_from_settings` we probably want to copy every WORKER_CLASS to the child